pkg/connect/routing/router.go

```diff
@@ -121,9 +140,25 @@ func getSuitableConnection(ctx context.Context, rnd *util.FrandRNG, stateMgr sta
 	}
 
 	healthy := make([]connWithGroup, 0, len(conns))
```

Comment:
```suggestion
	candidates := make([]connWithGroup, 0, len(conns))
```

Since we're using this to represent connections which are both 1) healthy and 2) have capacity, I think we can rename this to avoid confusion. Then your variables below make perfect sense 👍